synapse/lib/cell.py

```diff
@@ -503,17 +517,20 @@ async def __anit__(self, dirn, conf=None, readonly=False, *args, **kwargs):
 
         # generate a guid file if needed
         if not os.path.isfile(path):
+
             self.inaugural = True
+
+            guid = conf.get('cell:guid')
+            if guid is None:
+                guid = s_common.guid()
+
             with open(path, 'w') as fd:
-                fd.write(s_common.guid())
+                fd.write(guid)
 
         # read our guid file
         with open(path, 'r') as fd:
             self.iden = fd.read().strip()
 
-        if conf is None:
-            conf = {}
-
         self.conf = self._initCellConf(conf)
```

Comment:
If we're moving this conf stuff, lets move the _initCellConf up to  after the self.inauguarl = False setting.
this way, if we have invalid configuration, we don't make the cell.guid and avoid a mismatch in inaugural behaviors.

Reply:
ok